docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,187 @@
+* Guarantee of adaptability and ease of changing the design and implementation
+  in cases were customer needs evolve rapidly.
+* ConfigMaps are stored in the :term:`etcd` database which is generally being
+  backed up. This ensures that user settings cannot be lost easily.
+
+
+How it works
+~~~~~~~~~~~~
+
+Before Bootstrap, :file:`defaults.yaml` will be pre-filled with default service
+settings. Service pods and deployments will be configured to either
+consume configuration data directly from ConfigMaps or fallback to
+the default values if the service specific ConfigMap was not found.
+
+**Using Salt states**
+
+Once a ConfigMap is updated by the user (say a user changes the number of
+replicas for Prometheus deployments to a new value), then perform the
+following actions:
+
+  - Apply a Salt states that reads the ConfigMap object, validates the schema
+    and checks the new values passed.
+
+Storage format
+~~~~~~~~~~~~~~
+
+A sample ConfigMap can be defined with the following fields.
+
+An example of such a ConfigMap:
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      namespace: <namespace>
+      name: <config-name>
+    data:
+      config.yaml: |-
+        apiVersion: <object-version>
+        kind: <kind>
+        spec:
+          <key>: <values>
+
+**Use case 1:**
+
+Configure and store the number of replicas for service specific Deployments
+found in the `metalk8s-monitoring` namespace using the ConfigMap format.
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      namespace: metalk8s-monitoring
+      name: metalk8s-grafana-config
+    data:
+      config.yaml: |-
+        apiVersion: metalk8s.scality.com/v1alpha1
+        kind: GrafanaConfig
+        spec:
+          deployment:
+            replicas: 2
+
+Non-goals
+~~~~~~~~~
+
+This section contains requirements stated above which the current design choice
+does not cater for and will be addressed later:
+
+- Persisting newly added Grafana dashboards or new Grafana datasources
+  especially for modifications added via the Grafana UI cannot be stored in
+  ConfigMaps and hence will be catered for later.
+
+- As stated in the requirements, adding and editing Prometheus alert rules
+  is also not covered by the chosen design choice and will need to be addressed
+  differently.
+
+Rejected design choices
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Consul KV vs ConfigMap
+~~~~~~~~~~~~~~~~~~~~~~
+
+This approach offers a full fledge KV store with a /kv endpoint which allows
+CRUD operations on all KV data stored in it.
+Consul KV also allows access to past versions of objects and has an optimistic
+concurrency when manipulating multiple objects.
+
+Note that, Consul KV store was rejected because managing operations such as
+performing full backups, system restores for a full fledged KV system
+requires time and much more efforts than the ConfigMap approach.
+
+Operator (Custom Controller) vs ConfigMap
```

Comment:
`vs. Salt` - as Teddy said, Operator is not related to a given storage medium, and could work with plain CM